src/topics/models.py

```diff
@@ -49,13 +49,24 @@ def create_post(self, body: str, author_id: int) -> Post:
         return new_post
 
     @staticmethod
-    def get_topics(sorting: dict[str, str] | None = None) -> list[Topic]:
+    def get_topics(query_parameters: dict[str, list[Any]] | None = None) -> list[Topic]:
```

Comment:
Model's method should know nothing about the origin of the parameters. So we need to provide filtering data to model's method in a "ready" state. It should not be "query_parameters" because for the model it's irrelevant where did they come from. Model should not be responsible to parse and handle them. It accepts parameters which can used as is.

So you need change this method's signature to something like that:
```python
def get_topics(
    sorting: dict[str, str] | None = None,
    author_ids: list[int] | None = None,
    created_before: datetime | None = None,
    created_after: datetime | None = None,
) -> list[Topic]:
"""Use this method to get all topics from topics table."""
...
```
This will allow us to also use more strict type hints to our parameters. As you can see, instead of `dict[str, list[Any]]` we provided a concrete type for each parameter.

And after that you need to call it in `routes.py` in this way:
```python
@staticmethod
@authorized_access()
def get() -> list[dict[str, Any]] | None:
    """Get a sorted list of all topics."""
    parsed_args = reqparse_func()
    topics = Topic.get_topics(
        sorting=parsed_args.get("order_by"),
        author_ids=parsed_args.get("author_ids"),
        created_before=parsed_args.get("created_before"),
        created_after=parsed_args.get("created_after"),
    )
    ...
```